src/core/thread/network_diagnostic_tlvs.hpp

```diff
@@ -1181,7 +1189,7 @@ class ChildTableEntry
      * @returns The Child ID value.
      *
      */
-    uint16_t GetChildId(void) const { return HostSwap16(mTimeoutRsvChildId) & kChildIdMask; }
+    uint16_t GetChildId(void) const { return ((mFields.m8[0] & kChildIdMask) | mFields.m8[1]); }
```

Comment:
I agree with @jwhui however i am okay with the change.

Btw. I see on Travis (master), that the warning is actually:

```
./thread/network_diagnostic_tlvs.hpp:1228:42: warning: Function call argument is an uninitialized value
        mTimeoutRsvChildId = HostSwap16((HostSwap16(mTimeoutRsvChildId) & ~kReservedMask) |
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Wouldn't be enough to initialize a value of `mTimeoutRsvChildId `in the default constructor?